openedx/stanford/djangoapps/sneakpeek/templates/nonregistered/register-call-to-action.html

```diff
@@ -5,19 +5,16 @@
 from django.utils.translation import ugettext as _
 
 from openedx.core.djangolib.markup import HTML, Text
-
 from student.models import UserProfile
 %>
-% if course and user.is_authenticated() and not UserProfile.has_registered(user):
-  <p class="sneakpeek-warning">
+% if course and user.is_nonregistered():
   <%
   if settings.FEATURES.get('RESTRICT_ENROLL_BY_REG_METHOD'):
-    reg_url = reverse('course-specific-register', args=[course.id.to_deprecated_string()])
+    reg_url = reverse('course-specific-register', args=[unicode(course.id)])
   else:
     reg_url = reverse('register_user')
-  tos_link = u"<a href='{}'>".format(marketing_link('TOS'))
-  privacy_link = u"<a href='{}#privacy'>".format(marketing_link('TOS'))
   %>
+  <p class="sneakpeek-warning">
```

Comment:
Is there a reason you moved this after the conditional? This bit of HTML doesn't need any of the logic it's now following?

Reply:
> This bit of HTML doesn't need any of the logic it's now following?

Similarly, the page doesn't need that HTML fragment rendered before it's initialized all its variables ;)

> Is there a reason you moved this after the conditional?

Just putting the HTML together instead of splitting the markup between code blocks?